.github/workflows/linux-build-and-test.yml

```diff
@@ -0,0 +1,76 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: rclnodejs - Linux Build and Test
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build-humble-and-rolling: # humble and rolling distros
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [10.X, 12.x, 14.X, 16.X, 17.X]
+        ros_distribution:
```

Comment:
I have a question here, if we are submitting a PR targeting on `develop` branch and we only want to run the verification on the `rolling`, because sometimes the rolling may introduce some interface changes that will break the compiling on other releases. For the current configuration in this PR, it seems that all ROS2 distro will be selected?

Another case is when releasing a new version, we just want to run a specific branch on the corresponding ROS2 distro.

Reply:
Ah yes, this version of the linux-build-and-test.yml workflow is triggered by any pull_request or push events and runs build and test suite tasks for foxy, galactic, humble and rolling distros on node 10-17 (N*M=20 scenarios). Since we have been keeping rclnodejs backward compatible to foxy I went with this big workflow to identify compatibility issues back to foxy.

Proposal:
1 -  keep this workflow but convert it to only run manually. This will serve an all-in-one compatibility test workflow that we launch manually thru the github actions UI.
2 - impl a new workflow that is a 1-1 between branch and ros distro and triggered by pull_request and push events on the specific branch, e.g.,
| **BRANCH** | **ROS DISTRO** |
|:---:|:---:|
| develop | rolling |
| humble-hawksbill | humble |
| galactic-geochelone | galactic |
| foxy-fitzroy | foxy |
This new workflow would for example run the galactic-geochelone branch against only the galactic ros distro for PR and push events on the branch.

Thoughts?

OT - I have a windows version of the workflow running reliably on a branch that I'll revise to match our linux solution once complete.

Reply:
@minggangw [f1fd60a](https://github.com/RobotWebTools/rclnodejs/pull/860/commits/f1fd60a8280253a84e937ad733301e9e08d01c85) commit should address the requirements (I hope). 

I separated the linux and window ci into their own workflows. 

The following workflows are triggered by PR and push events on any of the foxy, galactic, humble or develop branches. They run ci for the changed branch against it's associated ros distro, i.e., foxy-fitzroy branch -> foxy distro.
windows-build-and-test.yml
linux-build-and-test.yml 

The following workflows are manually triggered and run a comprehensive ci of the `develop` branch against the foxy, galactic, humble and rolling distros:
windows-build-and-test-compatibility.yml
linux-build-and-test-compatibility.yml 

I was not able to fully tests these workflows in my dev env. (fingers crossed).

Also I doubt the workflows will work properly against the foxy and galactic branches without backporting the changes of this PR to those branches.